### PR TITLE
Cleanup API of HTTP and DNS

### LINKED ISSUE #44: Clean up the API of the HTTP library
* Clean up inconsitencies of API (use namespace instead of using prefix ight to class names). 

* Fix error codes to match OONI.

### PR DESCRIPTION
**Pull request by bassosimone**
Isolate the "good part" (a clean API) from the "bad part" (a messy implementation). First step towards the first release. Also, document couple of implementation bugs to be dealt with later.

Closes #44.

### CODE REVIEW

File: doc/api/c++/dns/resolver.md [hunk 1]
```diff
@@ -52,18 +52,40 @@ a resolver you can specify the following settings:
...
+(Of course, instead of specifying the types as strings, e.g. `"A"`, you
+can specify the corresponding query type, e.g. `QueryTypeId::A`.)
+
+The difference between `REVERSE_A`, `REVERSE_AAAA` and `PTR` is that
+`REVERSE_A` and `REVERSE_AAAA` take in input respectively an IPv4 and
+an IPv6 address, while for `PTR` you need to construct yourself the
+reversed IP address name to query.
+
+The callback could be called immediately if an error occurs while
+sending the query to the server.
```
Comment (hellais) [inline doc/api/c++/dns/resolver.md]: Nice docs!

File: include/measurement_kit/ooni/http_test.hpp [hunk 2]
```diff
@@ -51,7 +51,7 @@ class HTTPTest : public ooni::NetTest {
 
             if (error == 0) {
                 rr["response"]["headers"] = std::map<std::string, std::string>(response.headers);
-                rr["response"]["body"] = response.body.read();
+                rr["response"]["body"] = response.body;
```
Comment (hellais) [inline include/measurement_kit/ooni/http_test.hpp]: Lazy is always better. Thanks.

**Comment by hellais** [comment]
I tested this on Mac OS X 10.11.1 and it works. Great work!

```
============================================================================
Testsuite summary for measurement_kit 0.0.1
============================================================================
# TOTAL: 29
# PASS:  29
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
```

**Comment by bassosimone** [comment]
Thanks! :-)

**Pull request merged**